ui/app/scripts/controllers/newDeploymentCtrl.js

```diff
@@ -445,26 +614,113 @@ angular.module('apollo')
             return $scope.allDeployableVersions.filter(function(a){return a.gitCommitSha === sha})[0].id;
         }
 
-        // Data fetching
-		apolloApiService.getAllEnvironments().then(function(response) {
-            $scope.possibleEnvironments = response.data;
-			// Get selection from local storage
-            var previousEnvironmentIds = localStorageService.get(previouseEnvironmentLocalStorageKey);
+        apolloApiService.getAllEnvironments()
+         .then(parseAllEnvironments)
+         .then(() => {
+		     apolloApiService.getAllEnvironmentsStacks().then(function(response) {
+             var tempEnvironmentsStack = {};
+		     response.data.forEach(function(environmentsStack) {
+                tempEnvironmentsStack[environmentsStack.id] = environmentsStack;
+                addStackToDisplayEnvironments(environmentsStack);
+
+                var environmentsNames = [];
+                environmentsStack.environments.forEach(function(environmentId) {
+                    var environment = allEnvironments.find(x => x.id === environmentId);
+                    var environmentName = environment.name;
+                    environmentsNames = environmentsNames.concat(environmentName);
```

Comment:
Why? Isnt that enough that we add all names from all environments?
There is no way an environment could be in stack but not in allEnvironments

Reply:
I'm saving the environments names of every stack

Reply:
But you are doing that for all environments... It wont change in stacks